src/customerio-cdp.ts

```diff
@@ -1,39 +1,54 @@
-import { NativeModules, Platform } from 'react-native';
-import { CioLogLevel, type CioConfig } from './cio-config';
-import { type IdentifyParams } from './cio-params';
+import { type TurboModule } from 'react-native';
 import { CustomerIOInAppMessaging } from './customerio-inapp';
 import { CustomerIOPushMessaging } from './customerio-push';
 import { NativeLoggerListener } from './native-logger-listener';
+import {
+  CioLogLevel,
+  default as NativeModule,
+  type CioConfig,
+  type Spec as CodegenSpec,
+  type CustomAttributes,
+  type IdentifyParams,
+  type NativeSDKArgs,
+} from './specs/modules/NativeCustomerIO';
+import { callNativeModule, ensureNativeModule } from './utils/native-bridge';
+import { assert } from './utils/param-validation';
 
 const packageJson = require('customerio-reactnative/package.json');
 
-const LINKING_ERROR =
-  `The package 'customerio-reactnative' doesn't seem to be linked. Make sure: ` +
-  Platform.select({ ios: "- You have run 'pod install'\n", default: '' }) +
-  '- You rebuilt the app after installing the package\n' +
-  '- You are not using Expo Go\n';
-
-const NativeCustomerIO = NativeModules.NativeCustomerIO
-  ? NativeModules.NativeCustomerIO
-  : new Proxy(
-      {},
-      {
-        get() {
-          throw new Error(LINKING_ERROR);
-        },
-      }
+// Ensures all methods defined in codegen spec are implemented by the public module
+interface NativeSpec extends Omit<CodegenSpec, keyof TurboModule> {
+  isInitialized(): boolean;
+  inAppMessaging: CustomerIOInAppMessaging;
+  pushMessaging: CustomerIOPushMessaging;
+}
+
+// Track whether CustomerIO SDK has been initialized to prevent usage before setup
+let _initialized = false;
+
+// Reference to the native CustomerIO Data Pipelines module for SDK operations
+const nativeModule = ensureNativeModule(NativeModule);
+
+// Wrapper function that ensures SDK is initialized before calling native methods
+const withNativeModule = <R>(fn: (native: CodegenSpec) => R): R => {
+  if (!_initialized) {
+    throw new Error(
+      'CustomerIO SDK must be initialized before calling any methods. Please call CustomerIO.initialize() first.'
     );
+  }
+  return callNativeModule(nativeModule, fn);
+};
 
-export class CustomerIO {
-  private static initialized = false;
+export const CustomerIO = {
+  initialize: (config: CioConfig) => {
+    assert.config(config);
 
-  static readonly initialize = async (config: CioConfig) => {
     if (config.logLevel && config.logLevel !== CioLogLevel.None) {
       NativeLoggerListener.initialize();
     }
 
     const packageVersion = packageJson.version ?? '';
-    const args = {
+    const args: NativeSDKArgs = {
```

Comment:
can this work? might help reducing duplication 
```
const expoVersion = packageJson.expoVersion ?? '';
const args: NativeSDKArgs = {
  packageSource: expoVersion ? 'Expo' : 'ReactNative',
  packageVersion: expoVersion || packageJson.version || '',
};
```

Reply:
This looks good. Hopefully it doesn't break current Expo user agent 😅 will verify that before releasing, updated.